common/konfig.py

```diff
@@ -324,46 +317,21 @@ def profile_names(self) -> list[str]:
     def profile_ids(self) -> list[int]:
         return list(self._profiles.values())
 
-    def load(self):
+    def load(self, buffer: TextIOWrapper):
         """Load configuration from file like object."""
-        @contextlib.contextmanager
-        def _path_or_buffer(path_or_buffer: Union[Path, StringIO]
-                            ) -> Union[TextIOWrapper, StringIO]:
-            """Using a path or a in-memory file (buffer) with a with
-            statement."""
-            try:
-                # It is a regular file
-                path_or_buffer = path_or_buffer.open('r', encoding='utf-8')
-                print(f'{type(path_or_buffer)=}')
-
-            except AttributeError:
-                # Assuming a StringIO instance as in-memory file
-                pass
-
-            yield path_or_buffer
-
-            try:
-                # regular file: close it
-                path_or_buffer.close()
-
-            except AttributeError:
-                # in-memory file: "cursor" back to first byte
-                path_or_buffer.seek(0)
 
         self._config_parser = configparser.ConfigParser(
             interpolation=None,
             defaults={'profile1.name': _('Main profile')})
 
-        with _path_or_buffer(self._path) as handle:
-            print(handle)
-            content = handle.read()
-            logger.debug(f'Configuration read from "{self._path}".')
+        # raw content
+        content = buffer.read()
 
         # Add section header to make it a real INI file
         self._config_parser.read_string(f'{self._DEFAULT_SECTION}\n{content}')
 
         # The one and only main section
-        self._conf = self._config_parser['bit']
+        self._conf = self._config_parser[self._DEFAULT_SECTION[1:-1]]
```

Comment:
Do you mean `[1:]`?

``` python
a = [1, 2, 3, 4]
a[1:-1]
# [2, 3]
a[1:]
# [2, 3, 4]
```

Reply:
No. I just needed to cut of the square brackets from the string.

```
>>> _DEFAULT_SECTION = '[bit]'
>>> _DEFAULT_SECTION[1:-1]
'bit'
```